src/Avalonia.Base/Input/GestureRecognizers/ScrollGestureRecognizer.cs

```diff
@@ -181,53 +191,75 @@ protected override void PointerReleased(PointerReleasedEventArgs e)
                 else
                 {
                     _tracking = null;
-                    var savedGestureId = _gestureId;
-                    var st = Stopwatch.StartNew();
-                    var lastTime = TimeSpan.Zero;
-                    Target!.RaiseEvent(new ScrollGestureInertiaStartingEventArgs(_gestureId, _inertia));
-                    DispatcherTimer.Run(() =>
-                    {
-                        // Another gesture has started, finish the current one
-                        if (_gestureId != savedGestureId)
-                        {
-                            return false;
-                        }
-
-                        var elapsedSinceLastTick = st.Elapsed - lastTime;
-                        lastTime = st.Elapsed;
-
-                        var speed = _inertia * Math.Pow(InertialResistance, st.Elapsed.TotalSeconds);
-                        var distance = speed * elapsedSinceLastTick.TotalSeconds;
-                        var scrollGestureEventArgs = new ScrollGestureEventArgs(_gestureId, distance);
-                        Target!.RaiseEvent(scrollGestureEventArgs);
-
-                        if (!scrollGestureEventArgs.Handled || scrollGestureEventArgs.ShouldEndScrollGesture)
-                        {
-                            EndGesture();
-                            return false;
-                        }
-
-                        // EndGesture using InertialScrollSpeedEnd only in the direction of scrolling
-                        if (CanVerticallyScroll && CanHorizontallyScroll && Math.Abs(speed.X) < InertialScrollSpeedEnd && Math.Abs(speed.Y) <= InertialScrollSpeedEnd)
-                        {
-                            EndGesture();
-                            return false;
-                        }
-                        else if (CanVerticallyScroll && Math.Abs(speed.Y) <= InertialScrollSpeedEnd)
-                        {
-                            EndGesture();
-                            return false;
-                        }
-                        else if (CanHorizontallyScroll && Math.Abs(speed.X) < InertialScrollSpeedEnd)
-                        {
-                            EndGesture();
-                            return false;
-                        }
-
-                        return true;
-                    }, TimeSpan.FromMilliseconds(16), DispatcherPriority.Background);
+
+                    MediaContext.Instance.RequestAnimationFrame(StartInertia);
```

Comment:
You can use TopLevel.RequestAnimationFrame method, as it's a stable API. MediaContext is internal.

Reply:
Note that animation frames will be toplevel-scoped at some point too, the plan is to not trigger callbacks if toplevel isn't visible or is being throttled by the system compositor.

Reply:
Unfortunately, TopLevel is in the Controls project, and not the Base project.